[WIP] Add utils to visualize results with OpenCV

hetong007 opened the pull request:
- [X] Add opencv utils
- [x] Modify existing demos
- [ ] ~~Add classification demo~~
- [ ] ~~Add Semantic Segmentation demo~~
- [ ] ~~Add Instance Segmentation demo~~

mli commented:
Job PR-781-4 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-781/4/index.html 
Code coverage of this PR: ![pr.svg](http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-781/4/coverage.svg?) vs. Master: ![master.svg](http://gluon-cv.mxnet.io/coverage.svg?)

mli commented:
Job PR-781-5 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-781/5/index.html 
Code coverage of this PR: ![pr.svg](http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-781/5/coverage.svg?) vs. Master: ![master.svg](http://gluon-cv.mxnet.io/coverage.svg?)

hetong007 commented:
@zhreshold could you please review? I just realized that I have already implemented the `scale` parameter to allow plotting on the original resolution. Just removed too ambitious goals for this PR.

zhreshold commented on gluoncv/utils/viz/bbox.py:
```diff
@@ -102,3 +102,100 @@ def plot_bbox(img, bboxes, scores=None, labels=None, thresh=0.5,
+        bboxes[:, (0, 2)] *= width
+        bboxes[:, (1, 3)] *= height
+    else:
+        bboxes *= scale
```
only `True` if `absolute_coordinates=True`

hetong007 replied:
I couldn't get your comment here. Would you please explain more?

zhreshold commented on gluoncv/utils/viz/image.py:
```diff
@@ -38,3 +38,83 @@ def plot_image(img, ax=None, reverse_rgb=False):
+                                      bottomLeftCornerOfText[1] + 50)
+
+    canvas = cv2.cvtColor(img, cv2.COLOR_RGB2BGR)
+    cv2.imshow(canvas_name, canvas)
```
I am wondering if cv2.imshow and cv2.waitKey is appropriate here since it's not quite flexible to determine the wait time.

hetong007 replied:
I'm not quite certain about the role of this function. It is kind of specific to putting text at somewhere on the image. If we remove the `imshow`, it becomes a "cv_put_text" function.

zhreshold replied:
maybe we can leave the cv2.waitKey to users?

zhreshold replied:
just like we leave plt.show() to users

hetong007 replied:
Without `waitKey`, would `imshow` still trigger the plotting?

zhreshold replied:
Yes, but no window will pop out until you call cv2.waitKey

hetong007 replied:
OK that sounds fair.

mli commented:
Job PR-781-8 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-781/8/index.html 
Code coverage of this PR: ![pr.svg](http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-781/8/coverage.svg?) vs. Master: ![master.svg](http://gluon-cv.mxnet.io/coverage.svg?)
